File: TypeScript/ExampleTests/production/AddressComponents.ts
export class AddressComponents {
  zip: string;
  streetName: string;
  city: string;
  preDirection: string;
  suffixDirection: string;
  state: string;
  suffixType: string;

  constructor(zip: string, streetName: string, city: string, preDirection: string, suffixDirection: string, state: string, suffixType: string) {
    this.zip = zip;
    this.streetName = streetName;
    this.city = city;
    this.preDirection = preDirection;
    this.suffixDirection = suffixDirection;
    this.state = state;
    this.suffixType = suffixType;
  }
}

File: Rust/ExampleTests/production/addresscomponents.rs
use super::*;

#[derive(Debug, Clone, PartialEq)]
pub struct AddressComponents {
    pub zip: String,
    pub streetname: String,
    pub city: String,
    pub predirection: String,
    pub suffixdirection: String,
    pub state: String,
    pub suffixtype: String,
}

impl AddressComponents {
    pub fn new(zip: String, streetname: String, city: String, predirection: String, suffixdirection: String, state: String, suffixtype: String) -> Self {
        Self { zip,  streetname,  city,  predirection,  suffixdirection,  state,  suffixtype }
    }
}
